arch/x86/kvm/vmx/pkvm_init.c

```diff
@@ -683,13 +683,7 @@ int __init vmx_pkvm_init(void)
 	}
 
 	pr_info("All cpus are in guest mode!\n");
-	/*
-	 * TODO: Return -EFAULT to abort KVM init if the host has been
-	 * successfully deprivileged to prevent the host using vmx
-	 * instructions which are not supported by the pkvm hypervisor
-	 * until the pvVMCS is added.
-	 */
-	return -EFAULT;
+	return 0;
```

Comment:
Another thought: since we already have an implementation of re-privileging in https://android-review.googlesource.com/c/kernel/common/+/3649034, would it make sense to backport it to these early patches? So that if we've deprivileged, we _always_ re-privilege and return an error (and thus gracefully fallback to normal KVM, i.e. let KVM successfully initialize and run VMs as usual), until we implement pvVMCS. So that we don't regress KVM even temporarily.

It's not probably not the most critical thing to do now, but it makes sense at least in term of the order of patches?

@vineethrp for visibility.